[IV-24-25] Objetivo 2

pablotl0 opened the pull request:
Sobre la estructura del repositorio

* [x] ¿He seguido las mejores prácticas en el nombre de las clases y ficheros y
   disposición de los mismos?
* [x] ¿He mirado tanto reglas generales (para cualquier repo) como reglas
específicas para el lenguaje con el que se está trabajando?

Sobre el planteamiento

* [x] ¿Se han planteado una serie de issues, asignados a las historias de
usuario pertinentes, que clarifiquen qué es lo que necesitan los usuarios e
identifiquen las diferentes partes del problema?
* [x] ¿Los issues representan un problema, y no una tarea?

Sobre el análisis del problema

* [x] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo
   que se ha creado?
* [x] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor,
   una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo
   objeto valor que no dependa de ningún otro (y que sea la base de muchos
   otros)?.

Sobre el código

* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con a una HU?
* [x] ¿Ha asignado el propietario/la propietaria del repositorio todos los
issues planteados al *milestone* correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto
   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
   qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?


Lista de comprobación para el propietario/a del repositorio

* [x] ¿Los issues que ha planteado la persona son efectivamente problemas y
evidencian comprensión de las historias de usuario? Si no es así, es mejor que
se le transmita antes de que empiece a crear cualquier código erróneo.
* [x] ¿Has comprobado que efectivamente el compañero/a ha seguido *su* lista de
      comprobación?
* [x] ¿El código creado es una modelización válida de las historias de usuario
      creadas por ti, y te sirve como base para programar la lógica de negocio?
* [] ¿El código creado sigue las mejores prácticas para el lenguaje de
      programación que hayáis consensuado, tanto en identificadores, como
      nombres de ficheros, como disposición de los diferentes ficheros en el
      repositorio? Por ejemplo, ¿todos los commits se refieren a un issue y
      aportan código para resolver el problema planteado en el mismo?
* [x] ¿He comprobado que el código compile, o que quien lo ha programado ha lo
ha hecho y documentado en el repositorio?
* [x] ¿He añadido el PR y todos los issues que se resuelvan en él al milestone
correspondiente, que será el primero?
* [x] ¿Cubre totalmente el milestone 0, o hará falta modelización adicional en
el objetivo 4 para poder programar la lógica de negocio?
* [x] En resumen, ¿considero que con lo hecho puedo ponerme a trabajar
directamente en el siguiente milestone, que incluye la lógica de negocio?

pablotl0 committed:
closes #11, creando iv.yaml con el lenguaje acordado

pablotl0 committed:
closes #14,
creo el objeto valor Producto con el objetivo de representar un artículo. El objeto tiene un atributo 'nombre' para representar de manera básica el producto.

pablotl0 committed:
closes #13,
creo la entidad ElementoLista con el objetivo de representar los componentes de la lista de deseos. El objeto tiene un atributo 'producto' para representar de manera básica los componentes.

pablotl0 committed:
closes #12,
creo la entidad ListaDeseos con el objetivo de representar los regalos que un usuario quiere recibir. El objeto tiene un atributo 'deseos' para representar de manera básica el conjunto de regalos.

pablotl0 committed:
closes #16,
creo la entidad Usuario con el objetivo de representar unla persona que quiere recibir regalos. El objeto tiene unos atributos 'id', 'nombre', 'listaDeseos'  para representar de manera básica al usuario.

pablotl0 commented:
@evaanngiil Listo para revisión.

evaanngiil commented:
A la hora de crear un modelo no se deben poner funcionalidades ni funciones "main", únicamente la estructura básica y su constructor. Además de, sus posibles restricciones en los datos. @pablotl0

evaanngiil commented:
Por el principio cumple con los requerimientos del modelo. Sin embargo, me gustaría añadir un atributo a la entidad producto nombrada en #14. Cada producto, debería contener un atributo de precio, aparte del nombre.

pablotl0 committed:
Fix #14

En relación a https://github.com/evaanngiil/WishfulGiving/pull/17#issuecomment-2480564420 modifico la clase Producto para añadirle un precio al producto

pablotl0 commented:
@evaanngiil Ya está arreglado

evaanngiil commented:
Un último apunte, para seguir las buenas prácticas de la disposicion de los ficheros, creo que es conveniente que las entidades se encuentren en la carpeta única src.

pablotl0 committed:
Fix #11
